### PR TITLE
[MM-41981] Always install the latest compatible version via the Marketplace

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Always install the latest compatible version via the Marketplace. The tests in https://github.com/mattermost/mattermost-server/blob/94fb6e94c2eb9f975fd9566a4571bc35614ce8ae/api4/plugin_test.go#L1600-L1734 cover that case.

The logic changes in `app/plugin_install.go` ensure, that, if the prepacked version is old then the Marketplace one, the Marketplace one is used.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41981

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by srkgupta — APPROVED** [review]
Thanks @hanzei. Changes LGTM and is working fine.

**Comment by amyblais** [comment]
@catalintomai @levb This is a backport candidate.

File: app/plugin_install.go [hunk 1]
```diff
@@ -196,23 +196,39 @@ func (ch *Channels) InstallMarketplacePlugin(request *model.InstallMarketplacePl
...
+			if err != nil {
+				return nil, model.NewAppError("InstallMarketplacePlugin", "app.plugin.invalid_version.app_error", nil, "", http.StatusBadRequest)
+			}
 		}
-		signature, err := plugin.DecodeSignature()
+
+		marketplaceVersion, err := semver.Parse(plugin.Manifest.Version)
 		if err != nil {
-			return nil, model.NewAppError("InstallMarketplacePlugin", "app.plugin.signature_decode.app_error", nil, err.Error(), http.StatusNotImplemented)
+			return nil, model.NewAppError("InstallMarketplacePlugin", "app.plugin.invalid_version.app_error", nil, "", http.StatusBadRequest)
```
Comment (catalintomai) [inline app/plugin_install.go:217]: @hanzei, this error log entry is indistinguishable from the previous one, maybe we add smth. that denotes the differences(s)?

**Review by catalintomai — APPROVED** [review]

**Review by levb — APPROVED** [review]

File: app/plugin_install.go (hunk 1 above)
Comment (hanzei) [reply app/plugin_install.go]: Added a separate log message in 5a0fdba238c908c45c7d85099df40da4c0857ccd.

**Comment by hanzei** [comment]
@DHaussermann Could you please do some smoke testing around the Marketplace installed/update paths to confirm broke?

**Comment by amyblais** [comment]
@DHaussermann Also FYI, this is a backport candidate.

**Comment by DHaussermann** [comment]
@hanzei or @srkgupta Can you please advise on the testing scope for this? 
Should I just regression testing the UI marketplace?

To my understanding, the marketplace already installs the lates compatible version via the UI.

**Comment by srkgupta** [comment]
Yes @DHaussermann. Regression testing the plugin installation from marketplace should be sufficient.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Tested marketplace UI
- Version numbers listed are accurate
- Tested install process
- Tested with remote marketplace disabled
- No impacts or regressions found 

LGTM!

**Pull request merged**